### PR TITLE
Introduces CI/PR validation pipelines for all modules under sdk/

### PR DESCRIPTION
**Pull request by mitchdenny**
1. Tweaked the archetype template to use parameters instead of variables (better for reuse/composition in the future).
2. Added lots of ```ci.yml``` files.

I'm actually building everything under each SDK service folder and referencing it explicitly. Interested in your feedback here @scbedd since I'm explicitly referencing the modules that exist under each service folder.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by mitchdenny** [comment]
@scbedd I noticed that there is already a ```python - cosmos - ci``` pipeline definition. Once I merge ```sdk/cosmos/ci.yml``` and run my pipeline generation tool that definition will probably be overwritten with the stock standard one. Should the existing pipeline be moved/renamed?

**Comment by bryevdv** [comment]
Just chiming in to note that the Cosmos team has a separate v3 branch (that will be migrated/merged soon) and they will will need CI running against development on that branch for the foreseeable future.

**Comment by mitchdenny** [comment]
> ([that will be migrated/merged soon](https://github.com/Azure/azure-sdk-for-python/pull/5365))

We should cross check what the current pipeline is doing vs. the incoming pipeline. Looks like its structurally quite different in the way it tests compared with the rest of the SDK (spins up the emulator and then kicks off pytest).

@scbedd once you are back let's have a chat about how best to handle this variation (thanks for flagging it @bryevdv).

**Comment by mitchdenny** [comment]
```python - cosmos - ci``` has be renamed to ```python - cosmos - emulators``` to avoid conflict, but other than that they'll still kick off like normal. I'm now set to merge.

File: sdk/cognitiveservices/ci.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    include:
+    - master
+  paths:
+    include:
+    - sdk/cognitiveservices/
+
+jobs:
+- template: ../../eng/pipelines/templates/jobs/archetype-sdk-client.yml
+  parameters:
+    BuildTargettingString: azure-cognitiveservices-formrecognizer,azure-cognitiveservices-knowledge-nspkg,azure-cognitiveservices-language-nspkg,azure-cognitiveservices-nspkg,azure-cognitiveservices-search-nspkg,azure-cognitiveservices-vision-nspkg
```
Comment (scbedd) [inline sdk/cognitiveservices/ci.yml]: Is `BuildTargetingString` auto-generated based on file structure? We've been churning _super_ hard on the repo restructure, so I expect a bunch of these to change in the near future.

Comment (mitchdenny) [reply sdk/cognitiveservices/ci.yml]: Even though the file says that it is auto-generated - it isn't. So these are hand written. This file is becoming something of a customization configuration per pipeline where values that control behavior deeper in the templates are passed in.

We might remove the auto generated notice at the top of the file soon. The upshot is that these options are hand-written. If they aren't right we can change them.

Comment (scbedd) [reply sdk/cognitiveservices/ci.yml]: Ok! Sounds good.

**Comment by mitchdenny** [comment]
@scbedd could I get you to merge this one. The checks passed this time (not sure why I have to keep doing an update on the branch, the build duration makes it kinda hard to land anything on master).

**Comment by scbedd** [comment]
As long as there are no conflicts, we don't need to stay up with master. Merging.

**Pull request merged**